torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -616,7 +615,8 @@ std::shared_ptr<FutureMessage> TensorPipeAgent::send(
                          << clientPipe.pipe_->getRemoteName() << ": "
                          << error.what();
           }
-          markFutureWithError(std::move(futureResponseMessage), error.what());
+          auto it = clientPipe.pendingResponseMessage_.find(messageId);
+          markFutureWithError(it->second, error.what());
```

Comment:
I think you'll still need to check whether the future is in the map before marking with an error (doing `it->second` if the iterator has reached the end is undefined behavior I believe):

```
if (it != clientPipe.pendingResponseMessage_.end()) {
    markFutureWithError(it->second, error.what());
}
```

Reply:
I see, yeah this could have caused some unfortunate errors at runtime.